### DISCUSSION TITLE
Best practice to create a configuration file

### DISCUSSION THREAD

**Discussion by am000-simreka** [IDAES/idaes-pse#1489] [Q&A] [answered]
Hello, 
I am a new user of IDAES platform and I have to say I am impressed by all the capabilities and work that has been put into this software. My organization is very interested in expanding its use.
This question might seem a bit redundant but I want to know what is the best way to create or modify the configuration file to add different chemicals and their properties with the goal of using this in a property package in a process simulation.
Currently, I am taking the CO2_bmimPF6_PR.py from the examples as a template to rewrite a new config file for the Peng Robinson equation of state for a set of different chemicals (N-butane, N-pentane, and N-octane). However, I find it cumbersome to manually fill the data for entries such as critical properties and ideal gas heat capacity following the format of RPP4 or NIST. I would like to keep using these frameworks (RPP4, NIST, Perry, etc.) but fetching data form databases and avoid hard coding.
What is the standard way to create or modify the config file? The tutorials for “property package utilities” do not explain much about this

**Comment by andrewlee94** [comment, marked as answer]
Hi @am000-simreka, the short answer is that we do not have any standard utilities for creating property configuration file. This was partly due to not knowing what tools would be of most interest to users, and partly because many of these data sources are not in a form that is easily read (e.g. books). There is also the issue that it is often the case that no single database/source has all the data required. Thus, the "standard" way to create these configuration files is by manually collecting the data and writing the config dictionary by hand.

**Reply by am000-simreka** [reply]
Thank you for the reply @andrewlee94 . You may close this item.
